### DISCUSSION TITLE
Is it possible to load async specific namespace during SSR (without preloading)?

### DISCUSSION THREAD

**Discussion by knapcp3** [i18next/next-i18next#2168] [Q&A] [answered]
Hey,
I use `next-18next` with `i18next-http-backend` and what I want to achieve is loading the specific namespace asynchronously during server-side rendering of the page in which I use this namespace. So for example I have the `aboutTeamPage` namespace and inside the `getServerSideProps` of the `AboutTeam` page I call:
```
await serverSideTranslations('pl', ['aboutTeamPage'])
```
And I would like to load only this namespace (make the HTTP request for it) during server-side rendering of the `AboutTeam` page.
The `ns` option from the config file is empty to prevent preloading of the namespaces.

Is it possible to achieve this? Currently, if I do this, the namespace object inside the `initialI18nStore` object is empty, and the `i18next-http-backend` plugin and request are not fired during the rendering of the page. The `ready` flag returned from `useTranslation` is `true`  during SSR.

**Comment by adrai** [comment]
https://github.com/i18next/next-i18next#client-side-loading-of-translations-via-http

https://github.com/i18next/i18next-http-backend/tree/master/example/next#alternative-usage-the-preferred-way

**Reply by knapcp3** [reply]
@adrai thanks for the response. I'm aware I can lazy load translations on the client side, but is it possible on the server side to load the namespace that hasn't been loaded initially?
From what I understand and tested, if the config `ns` option looks like `ns: ['common']`, and then I do `await serverSideTranslations('pl', ['aboutTeamPage'])` inside `getServerSideProps`, the `aboutTeamPage` namespace won't be fetched at all. And I was wondering if I misconfigured something or if this is the current and intended behavior.

Basically why I'm asking this and what I worry about is that if I have loads of translation data in 20 languages, fetching it all during the first render of the server-side page rendering might slow it down significantly, because the request for initial translations data is done during the request time. Although I am aware that this will be slow only for the first request since then the translations are kept in memory.

**Reply by adrai** [reply, marked as answer]
please try with v13.3.0... that should also lazy load the namespaces on server side if using a custom backend

**Reply by knapcp3** [reply]
Works like a charm, appreciate it, thank you very much
